packages/nextjs-demo/src/app/asr/page.tsx

```diff
@@ -41,11 +63,12 @@ interface AsrProps {
 }
 
 const Asr: React.FC<AsrProps> = ({ name, link, id, costPerMinute, manager, transcript }) => {
-  const [output, setOutput] = useState('');
+  const [output, setOutput] = useState<Transcript[]>([]);
+  const outputRef = useRef(output);
```

Comment:
It feels weird to store the same value as a `ref` and as `state`, but not a huge deal.

Reply:
`state` is for things that impact the render result; `ref` is for everything else. In this case, it looks like `output` does eventually flow into the render result, so we'd use that and drop the `ref`.

Reply:
The real question here though is whether we should be rerendering (and reconnecting event listeners) on every new token. Plus I need the ref to ensure the div is scrolled to the bottom. Wdyt?

Reply:
Reconnecting event listeners on every new token doesn't seem right.

But overall, if things are working, I wouldn't worry more about this.

Reply:
don't we need to reconnect event listeners on every rerender, since the actual JS callback is new each time?